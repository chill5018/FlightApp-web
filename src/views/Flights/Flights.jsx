import React, { Component } from 'react';
import FlightCard from '../../components/FlightCard/index';


/* eslint-disable jsx-a11y/label-has-for */
class Flights extends Component {
  constructor(props) {
    super(props);
    this.state = {};

    this.bookFlight = this.bookFlight.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  componentDidMount() {
    const { flights, history } = this.props;

    if (!flights) {
      history.push('/');
    }
  }

  static getDerivedStateFromProps(props) {
    if (!props.flights) {
      props.history.push('/');
      return {};
    }

    return {
      flights: props.flights,
    };
  }

  handleChange = (e) => {
    e.preventDefault();

    const change = {};
    change[e.target.name] = e.target.value;

    this.setState(change);
  }

  bookFlight(event) {
    const { bookFlight } = this.props;

    bookFlight(event.target.value);
  }

  render() {
    const { flights } = this.state;
    return (
    <div className="app-main">
      <div className="app-main-wrapper">
        <div className="app-main-col_1">
         <p className="app-main-p">
          Total results:
          {' '}
          { flights.length }
         </p>
          { flights ? flights.map(flight => (
            <FlightCard key={flight.id} flight={flight} bookFlight={this.bookFlight} />
          )) : (
            <div>
              Loading...
            </div>
          )}
        </div>
      </div>
    </div>
    );
  }
}

export default Flights;
