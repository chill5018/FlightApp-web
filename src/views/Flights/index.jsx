import React from 'react';

class App extends React.Component {
  state = {}

  render() {
    return (
      <div>
        <div className="app-header">
          <h1>
            Purchasing
          </h1>
          <img className="app-header-image" src={logo} height="30px" width="auto" alt="logo"/>
        </div>

        <hr />

        <div className="app-content">
          <p>
            Show Available Flights
          </p>
        </div>
      </div>
    );
  }
}

export default App;
