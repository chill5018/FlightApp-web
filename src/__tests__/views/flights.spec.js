import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';

import Flights from '../../views/Flights';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Flight Snapshot', () => {
  it('renders correctly', () => {
    const initialState = {
      flights: {
        flights: undefined,
      },
    };

    const component = renderer.create(
    <Provider store={mockStore(initialState)}>
      <BrowserRouter>
        <Flights />
      </BrowserRouter>
    </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
