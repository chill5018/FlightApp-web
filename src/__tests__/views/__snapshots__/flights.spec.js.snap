// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flight Snapshot renders correctly 1`] = `
<div
  className="app-main"
>
  <div
    className="app-main-wrapper"
  >
    <div
      className="app-main-col_1"
    >
      <p
        className="app-main-p"
      >
        Total results:
         
        0
      </p>
    </div>
  </div>
</div>
`;
